app/services/structure_updater.rb

```diff
@@ -23,22 +23,36 @@ def initialize(model, csv)
   # @return [Bool] true if there are no problems
   def validate
     @errors = []
-    # Ensure all files in the csv are present in the existing object and ensure no existing files change preserve from no to yes
     csv.each.with_index(2) do |row, index|
-      if !existing_files_by_filename.key?(row['filename'])
-        errors << "On row #{index} found #{row['filename']}, which appears to be a new file"
-      elsif !existing_files_by_filename[row['filename']].administrative.sdrPreserve && row['preserve'] == 'yes'
-        errors << "On row #{index} found #{row['filename']}, which changed preserve from no to yes, which is not supported"
-      end
-      # ensure all supplied resource types are valid
-      unless Cocina::Models::FileSetType.properties.key?(row['resource_type'].to_sym)
-        errors << "On row #{index} found \"#{row['resource_type']}\", which is not a valid resource type"
-      end
+      errors << "On row #{index} found #{row['filename']}, which appears to be a new file" unless existing_file?(row)
```

Comment:
Is it possible to do `next` if the file doesn't exist? Then you don't need to repeat this check in all the other validations?

Reply:
I had the same thought and did it at first, but then you wouldn't get follow on errors for a missing file like the test expectation shows: https://github.com/sul-dlss/argo/blob/main/spec/services/structure_updater_spec.rb#L280   But perhaps that's not important and we can just adjust the test.

Reply:
What if we move the invalid_resource_type check up to the top?

Reply:
Yup, that works.  Adjusted the code and test (which now reverses the order of the error message).